tensorflow/core/kernels/mkl_fused_ops_test.cc

```diff
@@ -705,65 +743,25 @@ class FusedPadConvOpTest : public OpsTestBase {
     CommonTestUtilities<T> test_util;
-    test_util.ConvertAndCompare(dtype, first, second, expected);
+    test_util.ConvertAndCompareIntegral(dtype, first, second, expected);
   }
 };
 
-TEST_F(FusedPadConvOpTest, PaddingConvTest) {
-  const int depth = 1;
-  const int image_width = 4;
-  const int image_height = 3;
-  const int image_batch_count = 1;
-  Tensor image(DT_FLOAT, {image_batch_count, image_height, image_width, depth});
-  test::FillValues<float>(&image, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12});
+TYPED_TEST_CASE_P(FusedPadConvOpTest);
 
-  const int kFilterSize = 3;
-  const int kFilterCount = 1;
-  Tensor filter(DT_FLOAT, {kFilterSize, kFilterSize, depth, kFilterCount});
-  test::FillValues<float>(&filter, {1, 4, 7, 2, 5, 8, 3, 6, 9});
-
-  const int padding_height = 4;
-  const int padding_width = 2;
-  Tensor padding(DT_INT32, {padding_height, padding_width});
-  test::FillValues<int32>(&padding, {0, 0, 3, 4, 1, 2, 0, 0});
+TYPED_TEST_P(FusedPadConvOpTest, PaddingConvTest) { this->Run("NHWC"); }
 
-  Tensor expected(DT_FLOAT, TensorShape({1, 8, 5, 1}));
-  test::FillValues<float>(
-      &expected,
-      {0,  0,   0,   0,   0,   24, 42,  60,  33,  12,  105, 150, 183, 95,
-       32, 235, 312, 357, 178, 56, 187, 234, 261, 121, 32,  106, 126, 138,
-       59, 12,  0,   0,   0,   0,  0,   0,   0,   0,   0,   0});
+TYPED_TEST_P(FusedPadConvOpTest, PaddingConvTestNchw) { this->Run("NCHW"); }
 
-  Run<float>(DT_FLOAT, image, filter, padding, expected, "NHWC");
-}
-
-TEST_F(FusedPadConvOpTest, PaddingConvTestNchw) {
-  const int depth = 1;
-  const int image_width = 4;
-  const int image_height = 3;
-  const int image_batch_count = 1;
-  Tensor image(DT_FLOAT, {image_batch_count, depth, image_height, image_width});
-  test::FillValues<float>(&image, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12});
+REGISTER_TYPED_TEST_CASE_P(FusedPadConvOpTest, PaddingConvTest,
+                           PaddingConvTestNchw);
 
-  const int kFilterSize = 3;
-  const int kFilterCount = 1;
-  Tensor filter(DT_FLOAT, {kFilterSize, kFilterSize, depth, kFilterCount});
-  test::FillValues<float>(&filter, {1, 4, 7, 2, 5, 8, 3, 6, 9});
-
-  const int padding_height = 4;
-  const int padding_width = 2;
-  Tensor padding(DT_INT32, {padding_height, padding_width});
-  test::FillValues<int32>(&padding, {0, 0, 0, 0, 3, 4, 1, 2});
-
-  Tensor expected(DT_FLOAT, TensorShape({1, 1, 8, 5}));
-  test::FillValues<float>(
-      &expected,
-      {0,  0,   0,   0,   0,   24, 42,  60,  33,  12,  105, 150, 183, 95,
-       32, 235, 312, 357, 178, 56, 187, 234, 261, 121, 32,  106, 126, 138,
-       59, 12,  0,   0,   0,   0,  0,   0,   0,   0,   0,   0});
-
-  Run<float>(DT_FLOAT, image, filter, padding, expected, "NCHW");
-}
+#ifdef ENABLE_INTEL_MKL_BFLOAT16
+using FusedPadConvDataTypes = ::testing::Types<float, bfloat16>;
+#else
+using FusedPadConvDataTypes = ::testing::Types<float>;
+#endif
+INSTANTIATE_TYPED_TEST_CASE_P(Test, FusedPadConvOpTest, FusedPadConvDataTypes);
```

Comment:
```suggestion
INSTANTIATE_TYPED_TEST_SUITE_P(Test, FusedPadConvOpTest, FusedPadConvDataTypes);
```

Reply:
Thanks for review. All `TYPED_TEST_CASE_P` in this file have been replaced